include/mega/filefingerprint.h

```diff
@@ -59,4 +65,26 @@ struct MEGA_API FileFingerprintCmp
 
 bool operator==(const FileFingerprint& lhs, const FileFingerprint& rhs);
 
+// A light-weight fingerprint only based on size and mtime
+struct MEGA_API LightFileFingerprint
+{
+    m_off_t size = -1;
+    m_time_t mtime = 0;
+
+    LightFileFingerprint() = default;
+
+    MEGA_DEFAULT_COPY_MOVE(LightFileFingerprint)
+
+    // Generates a new fingerprint not involving I/O
+    bool genfingerprint(m_off_t filesize, m_time_t filemtime);
```

Comment:
I am not sure if the client needs to know what this function really does under the hood? Maybe later we'll change it to something else. Plus I think it's nice to have a consistent method name across the fingerprint classes. Speaking of which, maybe we could change all `genfingerprint` methods to be named just `from` or `make`?

Reply:
I like `from`, but since those are called with not too neat named variables (ll, f, fp, ...) I guess `genfingerprint` does give more information. Let's keep that one.